### ISSUE TITLE
[dtest]: nodetool_additional_test.TestNodetool.test_disablebinary_and_disablegossip failed on timeout

### ISSUE THREAD

**Issue by yaronkaikov** [scylladb/scylladb#16219]
Seen during the gating process - https://jenkins.scylladb.com/job/scylla-master/job/next/6874

```
self = <nodetool_additional_test.TestNodetool object at 0x7f777fe79dd0>

    def test_disablebinary_and_disablegossip(self):
        def run_stress(node, num_keys, mode, consistency, limited_rows_per_second=None):
            # mode = read or write
            logger.debug('Start stress command')
            rate_string = 'threads=10' + (f' throttle={limited_rows_per_second}/s' if limited_rows_per_second else '')
            result = node.stress([mode,
                                  f'n={num_keys}',
                                  f'cl={consistency}',
                                  '-rate', rate_string,
                                  '-schema', 'replication(factor=3)'],
                                 capture_output=True)
            logger.info('Stress results:\n' + format_cs_output(result))
            assert_cs_success(result)
    
        self.run_cluster(nodes=3)
        node1, node2, _ = self.cluster.nodelist()
        logger.info('Writing data')
        number_of_keys = 100000
        run_stress(node2, number_of_keys, mode="write", consistency="ALL")
        # Start stress in thread
        executor = ThreadPoolExecutor(max_workers=1)
        read_stress_run = executor.submit(run_stress,
                                          node=node2,
                                          num_keys=number_of_keys,
                                          mode="read",
                                          consistency="QUORUM",
                                          limited_rows_per_second=5000)
    
        for node in self.cluster.nodelist():
            node.flush()
        with nodetool_context(node=node1, start_command="disablebinary", end_command="enablebinary"):
            time.sleep(5)
            with nodetool_context(node=node1, start_command="disablegossip", end_command="enablegossip"):
                time.sleep(30)
                node1.compact()
        time.sleep(20)
        log_position = node1.mark_log()
        time.sleep(30)
        assert not node1.grep_log(expr="gate closed", from_mark=log_position), \
            "After executing enablebinary, the node still prints 'gate closed' messages"
        status = nodetool_status(node2)
        node1_status = None
        for s in status["nodes"]:
            if s['address'] == node1.address():
                node1_status = s['status']
                break
        assert node1_status, "{} not found in {}".format(node1.address(), status["nodes"])
        logger.info("Verifying node 1's status is UN")
        assert node1_status == "UN", "Node 1's status is expected to be UN, but instead it's {}".format(node1_status)
    
>       read_stress_run.result()

nodetool_additional_test.py:2687: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib64/python3.11/concurrent/futures/_base.py:451: in result
    self._condition.wait(timeout)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Condition(<unlocked _thread.RLock object owner=0 count=0 at 0x7f777ecd94c0>, 0)>
timeout = None

    def wait(self, timeout=None):
        """Wait until notified or until a timeout occurs.
    
        If the calling thread has not acquired the lock when this method is
        called, a RuntimeError is raised.
    
        This method releases the underlying lock, and then blocks until it is
        awakened by a notify() or notify_all() call for the same condition
        variable in another thread, or until the optional timeout occurs. Once
        awakened or timed out, it re-acquires the lock and returns.
    
        When the timeout argument is present and not None, it should be a
        floating point number specifying a timeout for the operation in seconds
        (or fractions thereof).
    
        When the underlying lock is an RLock, it is not released using its
        release() method, since this may not actually unlock the lock when it
        was acquired multiple times recursively. Instead, an internal interface
        of the RLock class is used, which really unlocks it even when it has
        been recursively acquired several times. Another internal interface is
        then used to restore the recursion level when the lock is reacquired.
    
        """
        if not self._is_owned():
            raise RuntimeError("cannot wait on un-acquired lock")
        waiter = _allocate_lock()
        waiter.acquire()
        self._waiters.append(waiter)
        saved_state = self._release_save()
        gotit = False
        try:    # restore state no matter what (e.g., KeyboardInterrupt)
            if timeout is None:
>               waiter.acquire()
E               Failed: Timeout >2700.0s

/usr/lib64/python3.11/threading.py:320: Failed
```

**Comment by fruch** [comment]
I'm introducing https://github.com/scylladb/scylla-dtest/pull/3757
so we'll have better information out of such a case, which seems like the c-s is stuck in the middle for some reason
now we'll have log of it (and a helper function to switch other places using c-s)

**Comment by patjed41** [comment]
Failed again in https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/5256/.

Logs (with logging for the c-s command enabled):
[dtest-gw1.log](https://github.com/scylladb/scylladb/files/13682980/dtest-gw1.log)
[1702583417114_nodetool_additional_test.py TestNodetool test_disablebinary_and_disablegossip.zip](https://github.com/scylladb/scylladb/files/13682982/1702583417114_nodetool_additional_test.py.TestNodetool.test_disablebinary_and_disablegossip.zip)

**Comment by Annamikhlin** [comment]
seen again in: https://jenkins.scylladb.com/job/scylla-master/job/next/7091/testReport/junit/nodetool_additional_test/TestNodetool/Build___x86___dtest___test_disablebinary_and_disablegossip/

**Comment by Annamikhlin** [comment]
seen again on:
https://jenkins.scylladb.com/job/scylla-master/job/next/7181/testReport/junit/nodetool_additional_test/TestNodetool/Build___x86___dtest___test_disablebinary_and_disablegossip/

**Comment by mykaul** [comment]
@fruch - can you look at the last failure and verify it's not the stress tool issue?

**Comment by fruch** [comment]
looks like the driver/c-s can't connect to any of the nodes, and then hang ~40min:
```
13:58:54,981 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: com.datastax.driver.core.exceptions.DriverInternalError: An unexpected error happened while sending requests
13:58:54,981 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: com.datastax.driver.core.exceptions.TransportException: [127.0.28.1/127.0.28.1:9042] Connection has been closed
13:58:54,981 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: com.datastax.driver.core.exceptions.TransportException: [127.0.28.1/127.0.28.1:9042] Connection has been closed
13:58:54,982 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: com.datastax.driver.core.exceptions.TransportException: [127.0.28.1/127.0.28.1:9042] Connection has been closed
13:58:55,996 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: WARN  13:58:55,993 Error creating netty channel to 127.0.28.1/127.0.28.1:9042
13:58:55,996 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: com.datastax.shaded.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: 127.0.28.1/127.0.28.1:9042
13:58:55,996 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: Caused by: java.net.ConnectException: Connection refused
13:58:55,997 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at java.base/sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
13:58:55,997 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:777)
13:58:55,997 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:337)
13:58:55,997 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334)
13:58:55,997 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:776)
13:58:55,997 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
13:58:55,997 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
13:58:55,997 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
13:58:55,997 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
13:58:55,997 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
13:58:55,997 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
13:58:55,997 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at java.base/java.lang.Thread.run(Thread.java:829)
13:58:58,001 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: WARN  13:58:58,000 Error creating netty channel to 127.0.28.1/127.0.28.1:9042
13:58:58,001 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: com.datastax.shaded.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: 127.0.28.1/127.0.28.1:9042
13:58:58,002 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: Caused by: java.net.ConnectException: Connection refused
13:58:58,002 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at java.base/sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
13:58:58,002 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:777)
13:58:58,002 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:337)
13:58:58,002 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334)
13:58:58,002 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:776)
13:58:58,002 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
13:58:58,002 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
13:58:58,002 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
13:58:58,002 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
13:58:58,002 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
13:58:58,002 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
13:58:58,002 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at java.base/java.lang.Thread.run(Thread.java:829)
13:59:00,003 797     ccm                            DEBUG    cluster.py          :762  | test_disablebinary_and_disablegossip: node1: nodetool cmd=disablegossip wait=True timeout=None
13:59:02,004 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: WARN  13:59:02,004 Error creating netty channel to 127.0.28.1/127.0.28.1:9042
13:59:02,005 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: com.datastax.shaded.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: 127.0.28.1/127.0.28.1:9042
13:59:02,005 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: Caused by: java.net.ConnectException: Connection refused
13:59:02,005 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at java.base/sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
13:59:02,005 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:777)
13:59:02,005 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:337)
13:59:02,005 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334)
13:59:02,005 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:776)
13:59:02,005 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
13:59:02,005 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
13:59:02,005 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
13:59:02,005 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
13:59:02,005 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
13:59:02,005 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
13:59:02,005 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at java.base/java.lang.Thread.run(Thread.java:829)
13:59:10,008 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: WARN  13:59:10,007 Error creating netty channel to 127.0.28.1/127.0.28.1:9042
13:59:10,008 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: com.datastax.shaded.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: 127.0.28.1/127.0.28.1:9042
13:59:10,008 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: Caused by: java.net.ConnectException: Connection refused
13:59:10,008 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at java.base/sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
13:59:10,008 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:777)
13:59:10,008 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:337)
13:59:10,008 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334)
13:59:10,008 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:776)
13:59:10,008 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
13:59:10,008 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
13:59:10,008 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
13:59:10,008 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
13:59:10,009 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
13:59:10,009 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
13:59:10,009 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at java.base/java.lang.Thread.run(Thread.java:829)
13:59:26,011 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: WARN  13:59:26,011 Error creating netty channel to 127.0.28.1/127.0.28.1:9042
13:59:26,011 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: com.datastax.shaded.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: 127.0.28.1/127.0.28.1:9042
13:59:26,011 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: Caused by: java.net.ConnectException: Connection refused
13:59:26,011 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at java.base/sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
13:59:26,011 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:777)
13:59:26,011 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:337)
13:59:26,011 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334)
13:59:26,011 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:776)
13:59:26,012 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
13:59:26,012 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
13:59:26,012 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
13:59:26,012 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
13:59:26,012 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
13:59:26,012 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at com.datastax.shaded.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
13:59:26,012 797     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: at java.base/java.lang.Thread.run(Thread.java:829)
```

reason is unclear from the logs

**Comment by Annamikhlin** [comment]
seen also here:
https://jenkins.scylladb.com/job/scylla-enterprise/job/next/2789/testReport/junit/nodetool_additional_test/TestNodetool/Build___x86___dtest___test_disablebinary_and_disablegossip/

**Comment by tchaikov** [comment]
again: https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/7873/artifact/logs-full.release.008/dtest-gw3.log

```
01:53:32,037 802     nodetool_additional_test       INFO     nodetool_additional_test.py:2566 | test_disablebinary_and_disablegossip: Verifying node 1's status is UN
02:36:25,398 802     errors                         ERROR    conftest.py         :202  | test_disablebinary_and_disablegossip: test failed: 
self = <nodetool_additional_test.TestNodetool object at 0x7f03a8e1db90>

    @pytest.mark.required_features("!consistent-topology-changes")
    def test_disablebinary_and_disablegossip(self):
        def run_stress(node, num_keys, mode, consistency, limited_rows_per_second=None):
...
            if timeout is None:
>               waiter.acquire()
E               Failed: Timeout >2700.0s
...
02:44:56,444 802     tools.stress                   DEBUG    stress.py           :98   | test_multidc_accuracy: c-s: at com.datastax.shaded.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
02:44:56,444 802     tools.stress                   DEBUG    stress.py           :98   | test_multidc_accuracy: c-s: at java.base/java.lang.Thread.run(Thread.java:829)
02:44:56,444 802     tools.stress                   DEBUG    stress.py           :98   | test_multidc_accuracy: c-s: WARN  02:44:56,440 Error creating netty channel to /127.0.53.1:9042
02:44:56,444 802     tools.stress                   DEBUG    stress.py           :98   | test_multidc_accuracy: c-s: com.datastax.shaded.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /127.0.53.1:9042
02:44:56,444 802     tools.stress                   DEBUG    stress.py           :98   | test_multidc_accuracy: c-s: Caused by: java.net.ConnectException: Connection refused
02:44:56,444 802     tools.stress                   DEBUG    stress.py           :98   | test_multidc_accuracy: c-s: at java.base/sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
02:44:56,444 802     tools.stress                   DEBUG    stress.py           :98   | test_multidc_accuracy: c-s: at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:777)
...
```
[1707878188420_nodetool_additional_test.py--TestNodetool--test_disablebinary_and_disablegossip.zip](https://github.com/scylladb/scylladb/files/14274838/1707878188420_nodetool_additional_test.py--TestNodetool--test_disablebinary_and_disablegossip.zip)

**Comment by nyh** [comment]
Failed again in CI run https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/6779/testReport/junit/nodetool_additional_test/TestNodetool/Tests___dtest___test_disablebinary_and_disablegossip/

**Comment by temichus** [comment]
Got the same issue in "raft topology mode" https://jenkins.scylladb.com/job/scylla-staging/job/artsiom_mishuta/job/dtest-pytest-gating/163/. 

it is fail 3 times of 100 runs

**Comment by temichus** [comment]
according to toscussion with @kbr and @fruch  I've created [PR](https://github.com/scylladb/scylla-dtest/pull/3992) to remove from next gating and assign issue  on @kostja

**Comment by Annamikhlin** [comment]
seen again on:
https://jenkins.scylladb.com/job/scylla-master/job/next/7306/testReport/junit/nodetool_additional_test/TestNodetool/Build___x86___dtest___test_disablebinary_and_disablegossip/

**Comment by avikivity** [comment]
Seen in unit tests too: https://github.com/scylladb/scylladb/issues/17353

**Comment by kbr-scylla** [comment]
> Seen in unit tests too: https://github.com/scylladb/scylladb/issues/17353

Different symptoms. In the unit test we're getting a query failure (driver returns NoHostAvailable). Here we see cassandra-stress hanging.

**Comment by denesb** [comment]
> > Seen in unit tests too: #17353
> 
> Different symptoms. In the unit test we're getting a query failure (driver returns NoHostAvailable). Here we see cassandra-stress hanging.

But the underlying issue might be the same: the java driver fumbles the re-connection to the cluster, after the quick disable/enable binary calls.

**Comment by avikivity** [comment]
@avelanarius we suspect a driver issue similar to that seen in the Python driver, please take a look

**Comment by fruch** [comment]
happened again 

and I noticed some weird print
```
15:41:44,603 722     tools.stress                   DEBUG    stress.py           :98   | test_disablebinary_and_disablegossip: c-s: com.datastax.driver.core.exceptions.DriverInternalError: An unexpected error happened while sending requests
```

it's coming from this function in the java-driver:
```java
    void findNextHostAndQuery() {
      try {
        Host host;
        while (!isDone.get()
            && (host = queryPlan.next()) != null
            && !queryStateRef.get().isCancelled()) {
          if (query(host)) {
            if (hostMetricsEnabled()) {
              metrics().getRegistry().counter(MetricsUtil.hostMetricName("writes.", host)).inc();
            }
            return;
          } else if (hostMetricsEnabled()) {
            metrics()
                .getRegistry()
                .counter(MetricsUtil.hostMetricName("write-errors.", host))
                .inc();
          }
        }
        if (current != null) {
          if (triedHosts == null) triedHosts = new CopyOnWriteArrayList<Host>();
          triedHosts.add(current);
        }
        reportNoMoreHosts(this);
      } catch (Exception e) {
        // Shouldn't happen really, but if ever the loadbalancing policy returned iterator throws,
        // we don't want to block.
        setFinalException(
            null,
            new DriverInternalError("An unexpected error happened while sending requests", e));
      }
    }
```

still doesn't fully explain why it's stuck

I did a PR to enable c-s extra debug, but it broke other tests, and we reverted it.

**Comment by fruch** [comment]
https://github.com/scylladb/scylla-dtest/pull/4039 is now in, and there at least one case of it failing with this extra debug:

https://jenkins.scylladb.com/job/scylla-enterprise/job/gating-dtest-release/1310/artifact/logs-full.release.007/dtest-gw1.log/*view*/

[dtest_log_with_full_cs_debug_16219.log](https://github.com/scylladb/scylladb/files/14550897/dtest_log_with_full_cs_debug_16219.log)

and we can see that driver reconnect o.k.
```
14:22:30,753 713     tools.stress                   DEBUG    stress.py           :102  | test_disablebinary_and_disablegossip: c-s: DEBUG 14:22:30,753 Added connection pool for /127.0.2.1:9042
```

but from some reason driver/c-s isn't executing any queries what so ever...

@avelanarius, we help with this one, we can't figure out why driver/c-s is hanging here.

**Comment by avikivity** [comment]
Seen again: https://jenkins.scylladb.com/view/nexts/job/scylla-enterprise/job/next/2841/testReport/junit/nodetool_additional_test/TestNodetool/Build___x86___dtest___test_disablebinary_and_disablegossip/

**Comment by fruch** [comment]
> Seen again: https://jenkins.scylladb.com/view/nexts/job/scylla-enterprise/job/next/2841/testReport/junit/nodetool_additional_test/TestNodetool/Build___x86___dtest___test_disablebinary_and_disablegossip/

It's the same one I was referring to 10m earlier (my link is from the dtest job, so it might be a bit misleading)

**Comment by avikivity** [comment]
@mykaul this one is hurting us

**Comment by yaronkaikov** [comment]
Again in https://jenkins.scylladb.com/job/scylla-5.4/job/next/217/

**Comment by mykaul** [comment]
> today we took it out of gating, @kostja promised to target looking into it for 6.0

@kostja - any updates?

**Comment by kostja** [comment]
https://github.com/scylladb/python-driver/issues/304 is fixed and merged, but https://github.com/scylladb/python-driver/issues/295 is not. Plus @kbr-scylla opened https://github.com/scylladb/python-driver/issues/317 today.

After the issues are closed,  a new driver release is made and the toolchain is updated, I'd look into this issue one more time.

**Comment by kbr-scylla** [comment]
This test doesn't use the Python driver, the problem is with a timing out cassandra-stress command, which uses Java driver underneath. (But we don't know if the problem is with cassandra-stress, Java driver, or Scylla)

Similar to https://github.com/scylladb/scylladb/issues/16002 by the way, there we also have cassandra-stress read timing out

**Comment by kbr-scylla** [comment]
6.0 is unrealistic. We have greater priorities.

**Comment by Annamikhlin** [comment]
seen again: https://jenkins.scylladb.com/job/enterprise-2024.1/job/next/162/testReport/junit/nodetool_additional_test/TestNodetool/Tests___dtest___test_disablebinary_and_disablegossip/

**Comment by Annamikhlin** [comment]
seen again on scylla-5.4 
https://jenkins.scylladb.com/job/scylla-5.4/job/next/280/testReport/junit/nodetool_additional_test/TestNodetool/Tests___dtest___test_disablebinary_and_disablegossip/

**Comment by mykaul** [comment]
> seen again on scylla-5.4 https://jenkins.scylladb.com/job/scylla-5.4/job/next/280/testReport/junit/nodetool_additional_test/TestNodetool/Tests___dtest___test_disablebinary_and_disablegossip/

Logs attached.
[test_disablebinary_and_disablegossip_node1.log](https://github.com/user-attachments/files/15545452/test_disablebinary_and_disablegossip_node1.log)
[test_disablebinary_and_disablegossip_node2.log](https://github.com/user-attachments/files/15545453/test_disablebinary_and_disablegossip_node2.log)
[test_disablebinary_and_disablegossip_node3.log](https://github.com/user-attachments/files/15545454/test_disablebinary_and_disablegossip_node3.log)
[test_disablebinary_and_disablegossip_dtest-gw3.log](https://github.com/user-attachments/files/15545455/test_disablebinary_and_disablegossip_dtest-gw3.log)

**Comment by mykaul** [comment]
2024.1.8 dtest-debug: https://jenkins.scylladb.com/job/enterprise-2024.1/job/dtest-debug/34/testReport/nodetool_additional_test/TestNodetool/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split030___test_disablebinary_and_disablegossip/

**Comment by mykaul** [comment]
2024.1.9/dtest-debug: https://jenkins.scylladb.com/job/enterprise-2024.1/job/dtest-debug/37/testReport/nodetool_additional_test/TestNodetool/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split030___test_disablebinary_and_disablegossip/

**Comment by emdroid** [comment]
Couldn't reproduce on my own yet, but having these observations so far:

At least the latest failures (the enterprise-2024.1 ones) are from the debug builds, and in these it looks like the first node appears to crash somehow - the log file ends prematurely (without a proper shutdown sequence, the last log being a compaction log, but no errors or backtraces).

So for example in the case of https://jenkins.scylladb.com/job/enterprise-2024.1/job/dtest-debug/37/testReport/nodetool_additional_test/TestNodetool/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split030___test_disablebinary_and_disablegossip/ :

Error from the test log (not a timeout):
```
12:41:48,605 705     errors                         ERROR    conftest.py         :226  | test_disablebinary_and_disablegossip: test failed: 
...
>           raise ToolError(cmd_args, rc, out, err)
E           ccmlib.node.ToolError: Subprocess ['stress', 'write', 'n=1000000', 'cl=ALL', '-rate', 'threads=10', '-schema', 'replication(factor=3)'] exited with non-zero status; exit status: 1; 
...
E           java.io.IOException: Operation x10 on key(s) [36383137395034364d30]: Error executing: (UnavailableException): Not enough replicas available for query at consistency ALL (3 required but only 2 alive)
```

This seems to be because the first node is dead for some reason, which is also reflected in the logs of the other nodes:
* node 2 (127.0.1.2):
```
ERROR 2024-08-30 12:41:32,660 [shard 1:stre] rpc - client 127.0.1.1:53955: server connection dropped: recv: Connection reset by peer
ERROR 2024-08-30 12:41:32,660 [shard 0:stre] rpc - client 127.0.1.1:7000: client connection dropped: recv: Connection reset by peer
ERROR 2024-08-30 12:41:32,690 [shard 1:stre] rpc - client 127.0.1.1:7000: client connection dropped: recv: Connection reset by peer
WARN  2024-08-30 12:41:32,694 [shard 1:stre] gossip - failure_detector_loop: Send echo to node 127.0.1.1, status = failed: seastar::rpc::closed_error (connection is closed)
WARN  2024-08-30 12:41:34,698 [shard 1:stre] gossip - failure_detector_loop: Send echo to node 127.0.1.1, status = failed: seastar::rpc::closed_error (connection is closed)
WARN  2024-08-30 12:41:36,700 [shard 1:stre] gossip - failure_detector_loop: Send echo to node 127.0.1.1, status = failed: seastar::rpc::closed_error (connection is closed)
Reactor stalled for 6838 ms on shard 0. Backtrace: 0xd955e8a 0x47666a4 0x4765bac 0x45a2a22 0x459d59f 0x459d047 0x459d908 0x45a3e4a 0x3dbaf 0xf648ee4 0xf646af1 0xf646740 0xf645f5c 0xf645bc7 0xf62e776 0xf608cc7 0xf51bd50 0xf51636f 0xdb98957 0xdb98630 0xdb9841d 0x45c4648 0x45cc7be 0x45d06a0 0x45ce373 0x413b24a 0x41387ea 0xd9f051a 0xd9ed008 0x27b89 0x27c4a 0xd910764
WARN  2024-08-30 12:41:38,703 [shard 1:stre] gossip - failure_detector_loop: Send echo to node 127.0.1.1, status = failed: seastar::rpc::closed_error (connection is closed)
WARN  2024-08-30 12:41:40,705 [shard 1:stre] gossip - failure_detector_loop: Send echo to node 127.0.1.1, status = failed: seastar::rpc::closed_error (connection is closed)
INFO  2024-08-30 12:41:40,813 [shard 1:comp] compaction - [Compact keyspace1.standard1 34905de0-66cd-11ef-9676-cacfeced32a2] Compacting [/jenkins/workspace/enterprise-2024.1/dtest-debug/scylla/.dtest/dtest-ev4k3ed8/test/node2/data/keyspace1/standard1-ec800fd066c911efa609ec00033a1180/me-3gj4_0yzl_4h6gw2adff8dlxlbj6-big-Data.db:level=0:origin=memtable,/jenkins/workspace/enterprise-2024.1/dtest-debug/scylla/.dtest/dtest-ev4k3ed8/test/node2/data/keyspace1/standard1-ec800fd066c911efa609ec00033a1180/me-3gj4_0z5u_3g3lc2adff8dlxlbj6-big-Data.db:level=0:origin=memtable,/jenkins/workspace/enterprise-2024.1/dtest-debug/scylla/.dtest/dtest-ev4k3ed8/test/node2/data/keyspace1/standard1-ec800fd066c911efa609ec00033a1180/me-3gj4_0ywr_2o8i82adff8dlxlbj6-big-Data.db:level=0:origin=memtable,/jenkins/workspace/enterprise-2024.1/dtest-debug/scylla/.dtest/dtest-ev4k3ed8/test/node2/data/keyspace1/standard1-ec800fd066c911efa609ec00033a1180/me-3gj4_0yjm_34ixc2adff8dlxlbj6-big-Data.db:level=0:origin=compaction]
WARN  2024-08-30 12:41:42,709 [shard 1:stre] gossip - failure_detector_loop: Send echo to node 127.0.1.1, status = failed: seastar::rpc::closed_error (connection is closed)
INFO  2024-08-30 12:41:42,709 [shard 1:stre] gossip - failure_detector_loop: Mark node 127.0.1.1 as DOWN
INFO  2024-08-30 12:41:42,711 [shard 0:stre] gossip - InetAddress 127.0.1.1 is now DOWN, status = NORMAL
INFO  2024-08-30 12:41:48,608 [shard 0:main] init - Signal received; shutting down

```
* node 3 (127.0.1.3):
```
Reactor stalled for 6626 ms on shard 1. Backtrace: 0xd955e8a 0x47666a4 0x4765bac 0x45a2a22 0x459d59f 0x459d047 0x459d908 0x45a3e4a 0x3dbaf 0xf648ee4 0xf646af1 0xf646740 0xf645f5c 0xf645bc7 0xf62e776 0xf608cc7 0xf51bd50 0xf51636f 0xdb98957 0xdb98630 0xdb9841d 0x45c4648 0x45cc7be 0x45d06a0 0x46cd1b7 0x46cb940 0x46cb830 0x46caf9c 0x439a8e8 0x8c946 0x11296f
ERROR 2024-08-30 12:41:32,660 [shard 0:stre] rpc - client 127.0.1.1:7000: client connection dropped: recv: Connection reset by peer
ERROR 2024-08-30 12:41:32,660 [shard 1:sl:d] rpc - client 127.0.1.1:7000: client connection dropped: recv: Connection reset by peer
ERROR 2024-08-30 12:41:32,663 [shard 1:stre] rpc - client 127.0.1.1:7000: client connection dropped: recv: Connection reset by peer
WARN  2024-08-30 12:41:32,668 [shard 1:stre] gossip - failure_detector_loop: Send echo to node 127.0.1.1, status = failed: seastar::rpc::closed_error (connection is closed)
WARN  2024-08-30 12:41:34,672 [shard 1:stre] gossip - failure_detector_loop: Send echo to node 127.0.1.1, status = failed: seastar::rpc::closed_error (connection is closed)
WARN  2024-08-30 12:41:36,675 [shard 1:stre] gossip - failure_detector_loop: Send echo to node 127.0.1.1, status = failed: seastar::rpc::closed_error (connection is closed)
WARN  2024-08-30 12:41:38,678 [shard 1:stre] gossip - failure_detector_loop: Send echo to node 127.0.1.1, status = failed: seastar::rpc::closed_error (connection is closed)
INFO  2024-08-30 12:41:40,549 [shard 1:comp] compaction - [Compact keyspace1.standard1 346703f0-66cd-11ef-a52f-ebff033a1180] Compacting [/jenkins/workspace/enterprise-2024.1/dtest-debug/scylla/.dtest/dtest-ev4k3ed8/test/node3/data/keyspace1/standard1-ec800fd066c911efa609ec00033a1180/me-3gj4_0yzl_15smo2iflmg9m3izy8-big-Data.db:level=0:origin=memtable,/jenkins/workspace/enterprise-2024.1/dtest-debug/scylla/.dtest/dtest-ev4k3ed8/test/node3/data/keyspace1/standard1-ec800fd066c911efa609ec00033a1180/me-3gj4_0yws_1sq8w2iflmg9m3izy8-big-Data.db:level=0:origin=memtable,/jenkins/workspace/enterprise-2024.1/dtest-debug/scylla/.dtest/dtest-ev4k3ed8/test/node3/data/keyspace1/standard1-ec800fd066c911efa609ec00033a1180/me-3gj4_0z5v_1s33k2iflmg9m3izy8-big-Data.db:level=0:origin=memtable,/jenkins/workspace/enterprise-2024.1/dtest-debug/scylla/.dtest/dtest-ev4k3ed8/test/node3/data/keyspace1/standard1-ec800fd066c911efa609ec00033a1180/me-3gj4_0yjo_2r8j42iflmg9m3izy8-big-Data.db:level=0:origin=compaction]
WARN  2024-08-30 12:41:40,680 [shard 1:stre] gossip - failure_detector_loop: Send echo to node 127.0.1.1, status = failed: seastar::rpc::closed_error (connection is closed)
WARN  2024-08-30 12:41:42,684 [shard 1:stre] gossip - failure_detector_loop: Send echo to node 127.0.1.1, status = failed: seastar::rpc::closed_error (connection is closed)
INFO  2024-08-30 12:41:42,684 [shard 1:stre] gossip - failure_detector_loop: Mark node 127.0.1.1 as DOWN
INFO  2024-08-30 12:41:42,686 [shard 0:stre] gossip - InetAddress 127.0.1.1 is now DOWN, status = NORMAL
INFO  2024-08-30 12:41:48,609 [shard 0:main] compaction_manager - Asked to stop
```

The last log line in the node 1 (127.0.1.1) is from 12:38:01:
```
INFO  2024-08-30 12:38:01,020 [shard 0:comp] compaction - [Compact keyspace1.standard1 b18f8830-66cc-11ef-8701-649a57e6902d] Compacting [/jenkins/workspace/enterprise-2024.1/dtest-debug/scylla/.dtest/dtest-ev4k3ed8/test/node1/data/keyspace1/standard1-ec800fd066c911efa609ec00033a1180/me-3gj4_0yqg_1vq9s21wrfhua8jnzh-big-Data.db:level=0:origin=memtable,/jenkins/workspace/enterprise-2024.1/dtest-debug/scylla/.dtest/dtest-ev4k3ed8/test/node1/data/keyspace1/standard1-ec800fd066c911efa609ec00033a1180/me-3gj4_0yzu_1c0e821wrfhua8jnzh-big-Data.db:level=0:origin=memtable,/jenkins/workspace/enterprise-2024.1/dtest-debug/scylla/.dtest/dtest-ev4k3ed8/test/node1/data/keyspace1/standard1-ec800fd066c911efa609ec00033a1180/me-3gj4_0yng_526n421wrfhua8jnzh-big-Data.db:level=0:origin=memtable,/jenkins/workspace/enterprise-2024.1/dtest-debug/scylla/.dtest/dtest-ev4k3ed8/test/node1/data/keyspace1/standard1-ec800fd066c911efa609ec00033a1180/me-3gj4_0ywr_067rk21wrfhua8jnzh-big-Data.db:level=0:origin=memtable]
```
and then just cut - looks like it could have been just killed (could be OOM because of debug?)

This then explains the error received ("`Not enough replicas available for query at consistency ALL (3 required but only 2 alive)`") as one of the nodes seems to be dead indeed.

The other case (enterprise detest-debug build 34) is basically the same.

The log files:
[node1.log](https://github.com/user-attachments/files/17078201/node1.log)
[node2.log](https://github.com/user-attachments/files/17078202/node2.log)
[node3.log](https://github.com/user-attachments/files/17078204/node3.log)
[dtest-gw2.log](https://github.com/user-attachments/files/17078207/dtest-gw2.log)

The abrupt termination without a stacktrace looks strange though, as under normal circumstances even on a crash I believe the stacktrace would be printed. Thus not sure what killed the node, but could be killed by the system (as mentioned, maybe OOM or something ... although the OOM killer would generally choose the victim randomly AFAIK).

**Comment by emdroid** [comment]
As for the older occurrences, at least from the logs attached for one of the scylla 5.4 occurrences the cause looks different though, as that is apparently a release build and there is a timeout indeed:
```
            if timeout is None:
>               waiter.acquire()
E               Failed: Timeout >2700.0s
```

plus also all the nodes are shut down gracefully.

**Comment by emdroid** [comment]
However, for the release build case I tried to reproduce it in dtest-byo with 100 and then 500 runs on the current scylla master: https://jenkins.scylladb.com/job/scylla-master/job/byo/job/dtest-byo/585/parameters/ but all was green.

So not sure if that means something, but it could be that the initial timeout problem may have been fixed somehow in the meantime (as 500 repetitions succeeded without any error). We still apparently have some other issue ongoing with the debug builds though, but it might have different causes - will continue looking.

I also tried to check with dtest-byo for the debug mode (would be again interested to perform some repetitions to see the percentage of failures), but couldn't get that working, the job is failing when using "debug" instead of the "release" build mode (ex.: https://jenkins.scylladb.com/job/scylla-master/job/byo/job/byo_build_tests_dtest/2485/)

**Comment by emdroid** [comment]
Yeah, I'd expect so (or at least the call stack in the log), but the most recent failures artifacts (e.g. https://jenkins.scylladb.com/job/enterprise-2024.1/job/dtest-debug/37/artifact/logs-full.debug.030/) don't seem to have any coredump either ... and the node1 log is just simply cut off, just ends with neither a graceful shutdown nor an error/callstack.

**Comment by emdroid** [comment]
For the debug build I managed to trigger the OOM locally on a laptop (running 4 dtest runs in parallel - which might be different than how the stuff is run on the dtest Jenkins build nodes though).

It is indeed the case that the OOM killer appears to kill the processes somewhat randomly, but killed a scylla process eventually:
```console
Sep 25 10:31:49 emaskovsky-dev kernel: Out of memory: Killed process 12981 (dbus-broker) total-vm:5024kB, anon-rss:176kB, file-rss:1764kB, shmem-rss:0kB, UID:984 pgtables:48kB oom_score_adj:200
Sep 25 10:31:49 emaskovsky-dev kernel: Out of memory: Killed process 12980 (dbus-broker-lau) total-vm:9128kB, anon-rss:0kB, file-rss:984kB, shmem-rss:0kB, UID:984 pgtables:64kB oom_score_adj:200
Sep 25 10:31:49 emaskovsky-dev kernel: Out of memory: Killed process 2311 (systemd) total-vm:29472kB, anon-rss:768kB, file-rss:1452kB, shmem-rss:0kB, UID:1000 pgtables:100kB oom_score_adj:100
Sep 25 10:31:49 emaskovsky-dev kernel: Out of memory: Killed process 2192 (systemd) total-vm:29648kB, anon-rss:1004kB, file-rss:1328kB, shmem-rss:0kB, UID:984 pgtables:100kB oom_score_adj:100
Sep 25 10:31:49 emaskovsky-dev kernel: Out of memory: Killed process 2194 ((sd-pam)) total-vm:16504kB, anon-rss:1060kB, file-rss:2320kB, shmem-rss:0kB, UID:984 pgtables:64kB oom_score_adj:100
Sep 25 10:31:49 emaskovsky-dev kernel: Out of memory: Killed process 2314 ((sd-pam)) total-vm:16500kB, anon-rss:916kB, file-rss:1408kB, shmem-rss:0kB, UID:1000 pgtables:64kB oom_score_adj:100
Sep 25 10:31:49 emaskovsky-dev kernel: Out of memory: Killed process 13683 (scylla) total-vm:21479606524kB, anon-rss:2612296kB, file-rss:468kB, shmem-rss:0kB, UID:1000 pgtables:18276kB oom_score_adj:0
```

- the scylla process with id 13683 has been killed.

Then looked for that process in the dtest logs:
```log
10:21:14,298 ccm             INFO  cluster.py     :757  | node1: Started scylla: pid: 13683
10:21:14,298 ccm             DEBUG cluster.py     :754  | node1: Starting scylla-jmx: args=['/home/emaskovsky/.dtest/dtest-szu8qi9u/test/node1/bin/symlinks/scylla-jmx', '-Dapiaddress=127.0.70.1', '-Djavax.management.builder.initial=com.scylladb.jmx.utils.APIBuilder', '-Djava.rmi.server.hostname=127.0.70.1', '-Dcom.sun.management.jmxremote', '-Dcom.sun.management.jmxremote.host=127.0.70.1', '-Dcom.sun.management.jmxremote.port=7199', '-Dcom.sun.management.jmxremote.rmi.port=7199', '-Dcom.sun.management.jmxremote.local.only=false', '-Xmx256m', '-XX:+UseSerialGC', '-Dcom.sun.management.jmxremote.authenticate=false', '-Dcom.sun.management.jmxremote.ssl=false', '-jar', '/home/emaskovsky/.dtest/dtest-szu8qi9u/test/node1/bin/scylla-jmx-1.0.jar']
```

so this is the first node there (of 3).

Checking the log of that particular node, there is indeed an abrupt end of the log (ending with a compaction log and nothing else):
```log
INFO  2024-09-25 10:31:43,902 [shard 0:strm] api - force_compaction: flush=true consider_only_existing_data=false
INFO  2024-09-25 10:31:43,917 [shard 0:strm] compaction_manager - User initiated compaction started on behalf of keyspace1.standard1 compaction_group=0
INFO  2024-09-25 10:31:43,921 [shard 0:strm] compaction - [Compact keyspace1.standard1 98776dd0-7b18-11ef-a1b9-55aa6f7ea2dd] Compacting [/home/emaskovsky/.dtest/dtest-szu8qi9u/test/node1/data/keyspace1/standard1-5dca2d907b1711efa1b955aa6f7ea2dd/me-3gju_0nd1_2hdlc2gjcflwmy19zx-big-Data.db:level=0:origin=memtable]
INFO  2024-09-25 10:31:43,923 [shard 1:strm] compaction - [Compact keyspace1.standard1 98776dd0-7b18-11ef-a74a-55a96f7ea2dd] Compacting [/home/emaskovsky/.dtest/dtest-szu8qi9u/test/node1/data/keyspace1/standard1-5dca2d907b1711efa1b955aa6f7ea2dd/me-3gju_0nd1_2e5uo2jl1ucd05j7q5-big-Data.db:level=0:origin=memtable]
```
the other logs (node2/3) have a graceful shutdown logged.

Also didn't find any  coredump for the process (there are some core dumps in the "/var/lib/systemd/coredump/" dir but none of them is from a scylla process) - it looks like the OOM killer doesn't produce coredumps by default indeed (although there seem to be way to enable it in the system).

So the above is consistent with the behavior we see in the dtest-debug builds. However to be completely sure we'd need to also gather the "/var/log/messages" log file (which is not currently being done) to verify that the process has indeed been OOM killed.

**Comment by emdroid** [comment]
Moved the debug build investigation to a new ticket  #20884 as it has a different reason (nodes being OOM killed - confirmed)

**Comment by mykaul** [comment]
2024.1.11/dtest-debug (https://jenkins.scylladb.com/job/enterprise-2024.1/job/dtest-debug/46/artifact/logs-full.debug.027/1727560899875_nodetool_additional_test.py%3A%3ATestNodetool%3A%3Atest_disablebinary_and_disablegossip/node1.log ):
```
INFO  2024-09-28 21:49:29,024 [shard 1:comp] compaction - [Compact keyspace1.standard1 89838200-7de3-11ef-baab-f572b9f7e9d3] Compacting [/jenkins/workspace/enterprise-2024.1/dtest-debug/scylla/.dtest/dtest-635pv2t4/test/node1/data/keyspace1/standard1-5300a7007de111efb13af573b9f7e9d3/me-3gjx_1og9_2jir42u71265fdsjbn-big-Data.db:level=0:origin=memtable,/jenkins/workspace/enterprise-2024.1/dtest-debug/scylla/.dtest/dtest-635pv2t4/test/node1/data/keyspace1/standard1-5300a7007de111efb13af573b9f7e9d3/me-3gjx_1odn_5nm8w2u71265fdsjbn-big-Data.db:level=0:origin=memtable]
Reactor stalled for 6683 ms on shard 0. Backtrace: 0xd971e8a 0x47666a4 0x4765bac 0x45a2a22 0x459d59f 0x459d047 0x459d908 0x45a3e4a 0x3dbaf 0xf665e24 0xf663a31 0xf663680 0xf662e9c 0xf662b07 0xf64b6b6 0xf625c07 0xf538c90 0xf5332af 0xdbb47c7 0xdbb44a0 0xdbb428d 0x45c4648 0x45cc7be 0x45d06a0 0x45ce373 0x413b24a 0x41387ea 0xda0c51a 0xda09008 0x27b89 0x27c4a 0xd92c764
WARN  2024-09-28 21:56:15,727 [shard 0:stre] gossip - failure_detector_loop: Send echo to node 127.0.30.3, status = failed: seastar::rpc::timeout_error (rpc call timed out)
INFO  2024-09-28 21:56:16,613 [shard 0:stre] gossip - failure_detector_loop: Mark node 127.0.30.3 as DOWN
WARN  2024-09-28 21:56:57,161 [shard 0:main] service_level_controller - update_from_distributed_data: failed to update configuration for more than  90 seconds : exceptions::read_timeout_exception (Operation timed out for system_distributed.service_levels - received only 0 responses from 1 CL=ONE.)
INFO  2024-09-28 21:57:03,499 [shard 0:stre] gossip - InetAddress 127.0.30.3 is now DOWN, status = NORMAL
INFO  2024-09-28 21:57:22,650 [shard 0:main] reader_concurrency_semaphore - Semaphore system with 2/10 count and 32768/21474836 memory resources: timed out, dumping permit diagnostics:
permits	count	memory	table/operation/state
1	1	16K	system_distributed.service_levels/data-query/waiting_for_execution
1	0	0B	system.batchlog/multishard-mutation-query/active
1	0	0B	*.*/view_builder/active

3	1	16K	total

Stats:
permit_based_evictions: 1
time_based_evictions: 0
inactive_reads: 1
total_successful_reads: 767
total_failed_reads: 0
total_reads_shed_due_to_overload: 0
total_reads_killed_due_to_kill_limit: 0
reads_admitted: 2549
reads_enqueued_for_admission: 528
reads_enqueued_for_memory: 0
reads_admitted_immediately: 2021
reads_queued_because_ready_list: 67
reads_queued_because_need_cpu_permits: 460
reads_queued_because_memory_resources: 1
reads_queued_because_count_resources: 0
reads_queued_with_eviction: 1
total_permits: 4319
current_permits: 4
need_cpu_permits: 0
awaits_permits: 0
disk_reads: 0
sstables_read: 0
INFO  2024-09-28 21:57:40,132 [shard 0:goss] gossip - InetAddress 127.0.30.3 is now UP, status = NORMAL
```

**Comment by mykaul** [comment]
Stall from above comment, analyzed:
```
Command line options: --address-threshold=0x100000000 --executable=/data/scylla-s3-reloc.cache/by-build-id/067b03fac87613c899ec79ebe451083d15e67c49/extracted/scylla/libexec/scylla --width=0 --direction=bottom-up --tmin=0 --branch-threshold=0.03 --file=<_io.TextIOWrapper name='<stdin>' mode='r' encoding='utf-8'>

Processed 1 stalls lasting a total of 6683 milliseconds.
min=6683 avg=6683.0 median=6683 p95=None p99=None p999=None max=6683

This graph is printed in bottom-up order, where callees are printed first.
Use --direction=top-down to print callers first.

[level#index/out_of pct%] below denotes:
  level  - nesting level in the graph
  index  - index of node among to its siblings
  out_of - number of siblings
  pct    - percentage of total stall time of this call relative to its siblings

++[0#1/1 100%] addr=0xd971e8a total=6683 count=1 avg=6683:
|              __interceptor_backtrace at ??:?
++           - addr=0x47666a4: ?? ??:0
++           - addr=0x4765bac: ?? ??:0
++           - addr=0x45a2a22: ?? ??:0
++           - addr=0x459d59f: ?? ??:0
++           - addr=0x459d047: ?? ??:0
++           - addr=0x459d908: ?? ??:0
++           - addr=0x45a3e4a: ?? ??:0
++           - addr=0x3dbaf: ?? ??:0
++           - addr=0xf665e24:
|              std::__detail::_Hash_node<std::pair<void const* const, logalloc::region_sanitizer::allocation>, false>::_M_next at /usr/bin/../lib/gcc/x86_64-redhat-linux/13/../../../../include/c++/13/bits/hashtable_policy.h:381
|              (inlined by) std::_Hashtable<void const*, std::pair<void const* const, logalloc::region_sanitizer::allocation>, std::allocator<std::pair<void const* const, logalloc::region_sanitizer::allocation> >, std::__detail::_Select1st, std::equal_to<void const*>, std::hash<void const*>, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Prime_rehash_policy, std::__detail::_Hashtable_traits<false, false, true> >::_M_rehash_aux at /usr/bin/../lib/gcc/x86_64-redhat-linux/13/../../../../include/c++/13/bits/hashtable.h:2592
++           - addr=0xf663a31:
|              std::_Hashtable<void const*, std::pair<void const* const, logalloc::region_sanitizer::allocation>, std::allocator<std::pair<void const* const, logalloc::region_sanitizer::allocation> >, std::__detail::_Select1st, std::equal_to<void const*>, std::hash<void const*>, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Prime_rehash_policy, std::__detail::_Hashtable_traits<false, false, true> >::_M_rehash at /usr/bin/../lib/gcc/x86_64-redhat-linux/13/../../../../include/c++/13/bits/hashtable.h:2565
|              (inlined by) std::_Hashtable<void const*, std::pair<void const* const, logalloc::region_sanitizer::allocation>, std::allocator<std::pair<void const* const, logalloc::region_sanitizer::allocation> >, std::__detail::_Select1st, std::equal_to<void const*>, std::hash<void const*>, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Prime_rehash_policy, std::__detail::_Hashtable_traits<false, false, true> >::_M_insert_unique_node at /usr/bin/../lib/gcc/x86_64-redhat-linux/13/../../../../include/c++/13/bits/hashtable.h:2179
++           - addr=0xf663680:
|              std::_Hashtable<void const*, std::pair<void const* const, logalloc::region_sanitizer::allocation>, std::allocator<std::pair<void const* const, logalloc::region_sanitizer::allocation> >, std::__detail::_Select1st, std::equal_to<void const*>, std::hash<void const*>, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Prime_rehash_policy, std::__detail::_Hashtable_traits<false, false, true> >::_M_merge_unique<std::_Hashtable<void const*, std::pair<void const* const, logalloc::region_sanitizer::allocation>, std::allocator<std::pair<void const* const, logalloc::region_sanitizer::allocation> >, std::__detail::_Select1st, std::equal_to<void const*>, std::hash<void const*>, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Prime_rehash_policy, std::__detail::_Hashtable_traits<false, false, true> > > at /usr/bin/../lib/gcc/x86_64-redhat-linux/13/../../../../include/c++/13/bits/hashtable.h:1125
++           - addr=0xf662e9c:
|              std::unordered_map<void const*, logalloc::region_sanitizer::allocation, std::hash<void const*>, std::equal_to<void const*>, std::allocator<std::pair<void const* const, logalloc::region_sanitizer::allocation> > >::merge<std::hash<void const*>, std::equal_to<void const*> > at /usr/bin/../lib/gcc/x86_64-redhat-linux/13/../../../../include/c++/13/bits/unordered_map.h:824
|              (inlined by) logalloc::region_sanitizer::merge(logalloc::region_sanitizer&)::{lambda()#1}::operator() at ./utils/logalloc.cc:355
++           - addr=0xf662b07:
|              logalloc::region_sanitizer::run_and_handle_errors<logalloc::region_sanitizer::merge(logalloc::region_sanitizer&)::{lambda()#1}> at ./utils/logalloc.cc:272
++           - addr=0xf64b6b6:
|              logalloc::region_sanitizer::merge at ./utils/logalloc.cc:350
|              (inlined by) logalloc::region_impl::merge at ./utils/logalloc.cc:2256
++           - addr=0xf625c07:
|              logalloc::region::merge at ./utils/logalloc.cc:2503
++           - addr=0xf538c90:
|              row_cache::do_update<row_cache::update(row_cache::external_updater, replica::memtable&)::$_0>(row_cache::external_updater, replica::memtable&, row_cache::update(row_cache::external_updater, replica::memtable&)::$_0)::{lambda()#1}::operator() at ./row_cache.cc:1027
|              (inlined by) std::__invoke_impl<seastar::future<void>, row_cache::do_update<row_cache::update(row_cache::external_updater, replica::memtable&)::$_0>(row_cache::external_updater, replica::memtable&, row_cache::update(row_cache::external_updater, replica::memtable&)::$_0)::{lambda()#1}&> at /usr/bin/../lib/gcc/x86_64-redhat-linux/13/../../../../include/c++/13/bits/invoke.h:61
|              (inlined by) std::__invoke_r<seastar::future<void>, row_cache::do_update<row_cache::update(row_cache::external_updater, replica::memtable&)::$_0>(row_cache::external_updater, replica::memtable&, row_cache::update(row_cache::external_updater, replica::memtable&)::$_0)::{lambda()#1}&> at /usr/bin/../lib/gcc/x86_64-redhat-linux/13/../../../../include/c++/13/bits/invoke.h:114
|              (inlined by) std::_Function_handler<seastar::future<void> (), row_cache::do_update<row_cache::update(row_cache::external_updater, replica::memtable&)::$_0>(row_cache::external_updater, replica::memtable&, row_cache::update(row_cache::external_updater, replica::memtable&)::$_0)::{lambda()#1}>::_M_invoke at /usr/bin/../lib/gcc/x86_64-redhat-linux/13/../../../../include/c++/13/bits/std_function.h:290
++           - addr=0xf5332af:
|              std::function<seastar::future<void> ()>::operator() at /usr/bin/../lib/gcc/x86_64-redhat-linux/13/../../../../include/c++/13/bits/std_function.h:591
|              (inlined by) row_cache::do_update(row_cache::external_updater, std::function<seastar::future<void> ()>)::$_0::operator()<row_cache::external_updater, std::function<seastar::future<void> ()> >(row_cache::external_updater&, std::function<seastar::future<void> ()>&) const::{lambda(auto:1)#1}::operator()<seastar::semaphore_units<seastar::semaphore_default_exception_factory, std::chrono::_V2::steady_clock> >(seastar::semaphore_units<seastar::semaphore_default_exception_factory, std::chrono::_V2::steady_clock>)::{lambda()#1}::operator()()::{lambda()#2}::operator() at ./row_cache.cc:1456
|              (inlined by) seastar::futurize<seastar::future<void> >::invoke<row_cache::do_update(row_cache::external_updater, std::function<seastar::future<void> ()>)::$_0::operator()<row_cache::external_updater, std::function<seastar::future<void> ()> >(row_cache::external_updater&, std::function<seastar::future<void> ()>&) const::{lambda(auto:1)#1}::operator()<seastar::semaphore_units<seastar::semaphore_default_exception_factory, std::chrono::_V2::steady_clock> >(seastar::semaphore_units<seastar::semaphore_default_exception_factory, std::chrono::_V2::steady_clock>)::{lambda()#1}::operator()()::{lambda()#2}> at ././seastar/include/seastar/core/future.hh:2006
|              (inlined by) seastar::futurize_invoke<row_cache::do_update(row_cache::external_updater, std::function<seastar::future<void> ()>)::$_0::operator()<row_cache::external_updater, std::function<seastar::future<void> ()> >(row_cache::external_updater&, std::function<seastar::future<void> ()>&) const::{lambda(auto:1)#1}::operator()<seastar::semaphore_units<seastar::semaphore_default_exception_factory, std::chrono::_V2::steady_clock> >(seastar::semaphore_units<seastar::semaphore_default_exception_factory, std::chrono::_V2::steady_clock>)::{lambda()#1}::operator()()::{lambda()#2}> at ././seastar/include/seastar/core/future.hh:2037
|              (inlined by) row_cache::do_update(row_cache::external_updater, std::function<seastar::future<void> ()>)::$_0::operator()<row_cache::external_updater, std::function<seastar::future<void> ()> >(row_cache::external_updater&, std::function<seastar::future<void> ()>&) const::{lambda(auto:1)#1}::operator()<seastar::semaphore_units<seastar::semaphore_default_exception_factory, std::chrono::_V2::steady_clock> >(seastar::semaphore_units<seastar::semaphore_default_exception_factory, std::chrono::_V2::steady_clock>)::{lambda()#1}::operator() at ./row_cache.cc:1455
|              (inlined by) seastar::futurize<seastar::future<void> >::invoke<row_cache::do_update(row_cache::external_updater, std::function<seastar::future<void> ()>)::$_0::operator()<row_cache::external_updater, std::function<seastar::future<void> ()> >(row_cache::external_updater&, std::function<seastar::future<void> ()>&) const::{lambda(auto:1)#1}::operator()<seastar::semaphore_units<seastar::semaphore_default_exception_factory, std::chrono::_V2::steady_clock> >(seastar::semaphore_units<seastar::semaphore_default_exception_factory, std::chrono::_V2::steady_clock>)::{lambda()#1}&> at ././seastar/include/seastar/core/future.hh:2006
|              (inlined by) seastar::futurize_invoke<row_cache::do_update(row_cache::external_updater, std::function<seastar::future<void> ()>)::$_0::operator()<row_cache::external_updater, std::function<seastar::future<void> ()> >(row_cache::external_updater&, std::function<seastar::future<void> ()>&) const::{lambda(auto:1)#1}::operator()<seastar::semaphore_units<seastar::semaphore_default_exception_factory, std::chrono::_V2::steady_clock> >(seastar::semaphore_units<seastar::semaphore_default_exception_factory, std::chrono::_V2::steady_clock>)::{lambda()#1}&> at ././seastar/include/seastar/core/future.hh:2037
|              (inlined by) seastar::future<void>::then<row_cache::do_update(row_cache::external_updater, std::function<seastar::future<void> ()>)::$_0::operator()<row_cache::external_updater, std::function<seastar::future<void> ()> >(row_cache::external_updater&, std::function<seastar::future<void> ()>&) const::{lambda(auto:1)#1}::operator()<seastar::semaphore_units<seastar::semaphore_default_exception_factory, std::chrono::_V2::steady_clock> >(seastar::semaphore_units<seastar::semaphore_default_exception_factory, std::chrono::_V2::steady_clock>)::{lambda()#1}, seastar::future<void> >(row_cache::external_updater&&)::{lambda((auto:1&&)...)#1}::operator()<> at ././seastar/include/seastar/core/future.hh:1406
|              (inlined by) seastar::noncopyable_function<seastar::future<void> ()>::indirect_vtable_for<seastar::future<void>::then<row_cache::do_update(row_cache::external_updater, std::function<seastar::future<void> ()>)::$_0::operator()<row_cache::external_updater, std::function<seastar::future<void> ()> >(row_cache::external_updater&, std::function<seastar::future<void> ()>&) const::{lambda(auto:1)#1}::operator()<seastar::semaphore_units<seastar::semaphore_default_exception_factory, std::chrono::_V2::steady_clock> >(seastar::semaphore_units<seastar::semaphore_default_exception_factory, std::chrono::_V2::steady_clock>)::{lambda()#1}, seastar::future<void> >(row_cache::external_updater&&)::{lambda((auto:1&&)...)#1}>::call at ././seastar/include/seastar/util/noncopyable_function.hh:158
++           - addr=0xdbb47c7:
|              seastar::noncopyable_function<seastar::future<void> ()>::operator() at ././seastar/include/seastar/util/noncopyable_function.hh:215
|              (inlined by) std::__invoke_impl<seastar::future<void>, seastar::noncopyable_function<seastar::future<void> ()>&> at /usr/bin/../lib/gcc/x86_64-redhat-linux/13/../../../../include/c++/13/bits/invoke.h:61
|              (inlined by) std::__invoke<seastar::noncopyable_function<seastar::future<void> ()>&> at /usr/bin/../lib/gcc/x86_64-redhat-linux/13/../../../../include/c++/13/bits/invoke.h:96
|              (inlined by) std::invoke<seastar::noncopyable_function<seastar::future<void> ()>&> at /usr/bin/../lib/gcc/x86_64-redhat-linux/13/../../../../include/c++/13/functional:113
|              (inlined by) seastar::internal::future_invoke<seastar::noncopyable_function<seastar::future<void> ()>&, seastar::internal::monostate> at ././seastar/include/seastar/core/future.hh:1155
|              (inlined by) seastar::future<void>::then_impl_nrvo<seastar::noncopyable_function<seastar::future<void> ()>, seastar::future<void> >(seastar::noncopyable_function<seastar::future<void> ()>&&)::{lambda(seastar::internal::promise_base_with_type<void>&&, seastar::noncopyable_function<seastar::future<void> ()>&, seastar::future_state<seastar::internal::monostate>&&)#1}::operator()(seastar::internal::promise_base_with_type<void>&&, seastar::noncopyable_function<seastar::future<void> ()>&, seastar::future_state<seastar::internal::monostate>&&) const::{lambda()#1}::operator() at ././seastar/include/seastar/core/future.hh:1458
++           - addr=0xdbb44a0:
|              seastar::futurize<seastar::future<void> >::satisfy_with_result_of<seastar::future<void>::then_impl_nrvo<seastar::noncopyable_function<seastar::future<void> ()>, seastar::future<void> >(seastar::noncopyable_function<seastar::future<void> ()>&&)::{lambda(seastar::internal::promise_base_with_type<void>&&, seastar::noncopyable_function<seastar::future<void> ()>&, seastar::future_state<seastar::internal::monostate>&&)#1}::operator()(seastar::internal::promise_base_with_type<void>&&, seastar::noncopyable_function<seastar::future<void> ()>&, seastar::future_state<seastar::internal::monostate>&&) const::{lambda()#1}> at ././seastar/include/seastar/core/future.hh:1991
++           - addr=0xdbb428d:
|              seastar::future<void>::then_impl_nrvo<seastar::noncopyable_function<seastar::future<void> ()>, seastar::future<void> >(seastar::noncopyable_function<seastar::future<void> ()>&&)::{lambda(seastar::internal::promise_base_with_type<void>&&, seastar::noncopyable_function<seastar::future<void> ()>&, seastar::future_state<seastar::internal::monostate>&&)#1}::operator() at ././seastar/include/seastar/core/future.hh:1454
|              (inlined by) seastar::continuation<seastar::internal::promise_base_with_type<void>, seastar::noncopyable_function<seastar::future<void> ()>, seastar::future<void>::then_impl_nrvo<seastar::noncopyable_function<seastar::future<void> ()>, seastar::future<void> >(seastar::noncopyable_function<seastar::future<void> ()>&&)::{lambda(seastar::internal::promise_base_with_type<void>&&, seastar::noncopyable_function<seastar::future<void> ()>&, seastar::future_state<seastar::internal::monostate>&&)#1}, void>::run_and_dispose at ././seastar/include/seastar/core/future.hh:741
++           - addr=0x45c4648: ?? ??:0
++           - addr=0x45cc7be: ?? ??:0
++           - addr=0x45d06a0: ?? ??:0
++           - addr=0x45ce373: ?? ??:0
++           - addr=0x413b24a: ?? ??:0
++           - addr=0x41387ea: ?? ??:0
++           - addr=0xda0c51a:
|              scylla_main at ./main.cc:694
++           - addr=0xda09008:
|              std::function<int (int, char**)>::operator() at /usr/bin/../lib/gcc/x86_64-redhat-linux/13/../../../../include/c++/13/bits/std_function.h:591
|              (inlined by) main at ./main.cc:2114
++           - addr=0x27b89: ?? ??:0
++           - addr=0x27c4a: ?? ??:0
++           - addr=0xd92c764:
|              _start at ??:?
```

**Comment by kbr-scylla** [comment]
After investigation the c-s hanging issue is suspected to be a problem in drivers/cassandra-stress, likely with the same root cause as https://github.com/scylladb/scylladb/issues/16002 (https://github.com/scylladb/cassandra-stress/issues/30)

Probably best course of action is fixing https://github.com/scylladb/cassandra-stress/issues/30, then seeing if this issue keeps reproducing.

So marking as blocked by https://github.com/scylladb/cassandra-stress/issues/30, and handing over to drivers team
